elastic/kibana
# [Response Ops][Actions] Add ability to post Block Kit messages to Slack Web API action

Resolves https://github.com/elastic/kibana/issues/88832

## Summary

Updates the Slack Web API connector with option to enter Block Kit JSON in the action parameters, including rendering action variables within the JSON.

## To Verify

1. Create a Slack Web API connector
2. Create a rule that will fire notifications and add an action.
3. Select `Block Kit` from the button group and enter block kit JSON. This input should show a validation error for invalid JSON. You should also be able to use action variables
<img width="591" alt="Screenshot 2024-01-05 at 2 21 42 PM" src="https://github.com/elastic/kibana/assets/13104637/59779015-4e18-4fb1-b81b-526c1a593baa">

**Example Block Kit for ES Query Rule**
```
{
	"blocks": [
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "*Elasticsearch query rule '{{rule.name}}' is active at {{context.date}}*\n - Value: {{context.value}}\n - Conditions Met: {{context.conditions}} over {{rule.params.timeWindowSize}}{{rule.params.timeWindowUnit}}\n\n:notebook: <https://runbook.com|Runbook> - :chart_with_upwards_trend: <https://kibana.dashboard|Kibana> - :cloud: <https://admin|Admin> - :eyes: <https://monitoring|Monitoring>"
			},
			"accessory": {
				"type": "image",
				"image_url": "https://i.guim.co.uk/img/static/sys-images/Guardian/Pix/pictures/2009/4/29/1240996556472/exclamation-001.jpg?width=465&dpr=1&s=none",
				"alt_text": "exclamation"
			}
		}
	]
}
```


5. Use the link to validate the payload using Slack's block kit builder
6. Save the rule. When it runs, you should receive a slack notification with the correct formatting.
<img width="645" alt="Screenshot 2024-01-05 at 2 22 44 PM" src="https://github.com/elastic/kibana/assets/13104637/39d37857-e751-467e-be39-cba0ea5d8399">

## Note
Functional tests for creating a rule with a slack web API action are currently skipped so I did not update them in this PR. I created a follow-on issue https://github.com/elastic/kibana/issues/174398 because reviving those functional tests is more complicated than just unskipping them.
